src/main/java/seedu/address/model/task/DistinctTaskList.java

```diff
@@ -55,43 +55,34 @@ public void addTask(Task taskAdded) {
         taskList.add(taskAdded);
     }
 
-    /**
-     * Replaces the task {@code target} in the list with {@code editedTask}.
-     * {@code target} must exist in the list.
-     * The task identity of {@code editedTask} should be the same as task identity of {@code target}.
-     */
-    public void setTask(Task target, Task editedTask) {
-        requireAllNonNull(target, editedTask);
-
-        int index = taskList.indexOf(target);
-        if (index == -1) {
-            throw new TaskNotFoundException();
-        }
-
-        if (!target.isSameTask(editedTask)) {
-            throw new TaskIdentityModifiedException();
-        }
-
-        taskList.set(index, editedTask);
-    }
-
     /**
      * Replaces the given task {@code target} with {@code editedTask}.
      * {@code target} must exist in the task list.
+     * If {@code isSameTask} is true, the task identity of {@code editedTask} should be the same as {@code target}.
      *
-     * @throws DuplicateTaskException if task identity of {@code editedTask} is the same as another task
-     *     in the list (other than {@code target}).
+     * @param target the task to be replaced.
+     * @param editedTask the edited task to replace {@code target}.
+     * @param isSameTask true if {@code target} has the same task identity as {@code editedTask}, false otherwise.
+     * @throws DuplicateTaskException if {@code isSameTask} is false but task identity of {@code editedTask}
+     *     is the same as another task in the list (other than {@code target}).
      */
-    public void replaceTask(Task target, Task editedTask) throws DuplicateTaskException {
-        requireAllNonNull(target, editedTask);
+    public void replaceTask(Task target, Task editedTask, boolean isSameTask) throws DuplicateTaskException {
+        requireAllNonNull(target, editedTask, isSameTask);
 
         int index = taskList.indexOf(target);
         if (index == -1) {
             throw new TaskNotFoundException();
         }
-        if (contains(editedTask) && !editedTask.isSameTask(target)) {
+
+        if (isSameTask && !target.isSameTask(editedTask)) {
+            throw new WrongTaskModifiedException();
+        }
+
+        boolean isDuplicateTask = contains(editedTask) && !editedTask.isSameTask(target);
+        if (!isSameTask && contains(editedTask) && isDuplicateTask) {
```

Comment:
Spotted minor bug in code after merging where contains(ediedTask) here is unnecessary

Reply:
Thanks for spotting the bug, I have removed the extra 'contains'.